src/Symfony/Component/Yaml/Inline.php

```diff
@@ -104,6 +108,18 @@ public static function dump($value, $exceptionOnInvalidType = false, $objectSupp
 
                 return 'null';
             case is_object($value):
+                if ($value instanceof \DateTime || $value instanceof \DateTimeImmutable) {
```

Comment:
use ``DateTimeInterface`` rather than ``DateTimeImmutable``. This way, the DateTimeInterface check covers all supported cases for PHP 5.5+ (and we can drop the DateTime check in 3.0 when dropping the PHP 5.4 support)